test/libsolidity/GasTest.h

```diff
@@ -32,7 +32,7 @@
 namespace solidity::frontend::test
 {
 
-class GasTest: AnalysisFramework, public TestCase
+class GasTest: AnalysisFramework, public EVMVersionRestrictedTestCase
```

Comment:
If we're doing this, then we should also convert this bit: https://github.com/ethereum/solidity/blob/82cefdf3374ed610614054290933f6300265c6f8/test/libsolidity/GasTest.cpp#L118-L120

I.e. just remove it - `AnalysisFramework` by default configures the `ComilerStack` to use EVM and EOF versions from command-line. Then in the tests also include `EVMVersion: =current` to explicitly restrict them to the latest EVM version (unless they do work with a wider range of versions but I doubt it, and it's probably not worth it to bother checking).

Without it the values read from the test are actually ignored. I actually wonder why they even pass here given that only EVM version is being overridden and EOF is left unchanged.

Reply:
They did not pass before the change. Now they pass because this tests are omitted for EOF compiler flag. 
Applied the change you suggested.

Reply:
Ah, I somehow keep forgetting that EOF is not included by default :)